crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -155,6 +135,200 @@ impl FormatRule<Expr, PyFormatContext<'_>> for FormatExpr {
+/// This is independent of whether they are own line or end-of-line comments, though end-of-line
+/// comments can become own line comments when we discard nested parentheses.
+///
+/// Style decision: When there are multiple nested parentheses around an expression, we consider the
+/// outermost parentheses the relevant ones and discard the others.
+fn format_with_parentheses_comments(
+    expression: &Expr,
+    node_comments: &LeadingDanglingTrailingComments,
+    f: &mut PyFormatter,
+) -> FormatResult<()> {
+    // First part: Split the comments
+
+    // TODO(konstin): This is copied from `parenthesized_range`, except that we don't have the
+    // parent, which is a problem:
+    // ```python
+    // f(
+    //     # a
+    //     (a)
+    // )
+    // ```
+    // gets formatted as
+    // ```python
+    // f(
+    //     (
+    //         # a
+    //         a
+    //     )
+    // )
+    // ```
+    let right_tokenizer = SimpleTokenizer::starts_at(expression.end(), f.context().source())
+        .skip_trivia()
+        .take_while(|token| token.kind == SimpleTokenKind::RParen);
+
+    let left_tokenizer = BackwardsTokenizer::up_to(
+        expression.start(),
+        f.context().source(),
+        f.context().comments().ranges(),
+    )
+    .skip_trivia()
+    .take_while(|token| token.kind == SimpleTokenKind::LParen);
+
+    // Zip closing parenthesis with opening parenthesis. The order is intentional, as testing for
+    // closing parentheses is cheaper, and `zip` will avoid progressing the `left_tokenizer` if
+    // the `right_tokenizer` is exhausted.
+    let range_with_parens = right_tokenizer
+        .zip(left_tokenizer)
+        .last()
+        .map(|(right, left)| TextRange::new(left.start(), right.end()));
+
+    let (leading_split, trailing_split) = if let Some(range_with_parens) = range_with_parens {
+        let leading_split = node_comments
+            .leading
+            .partition_point(|comment| comment.start() < range_with_parens.start());
+        let trailing_split = node_comments
+            .trailing
+            .partition_point(|comment| comment.start() < range_with_parens.end());
+        (leading_split, trailing_split)
+    } else {
+        (0, node_comments.trailing.len())
+    };
+
+    let leading_outer = &node_comments.leading[..leading_split];
+    let leading_inner = &node_comments.leading[leading_split..];
+    let trailing_inner = &node_comments.trailing[..trailing_split];
+    let trailing_outer = &node_comments.trailing[trailing_split..];
+
+    // Preserve an opening parentheses comment
+    // ```python
+    // a = ( # opening parentheses comment
+    //     # leading inner
+    //     1
+    // )
+    // ```
+    let (parentheses_comment, leading_inner) = match leading_inner.split_first() {
+        Some((first, rest)) if first.line_position().is_end_of_line() => {
+            (slice::from_ref(first), rest)
+        }
+        _ => (Default::default(), node_comments.leading),
```

Comment:
I'm surprised clippy doesn't complain here. It normally doesn't like `Default::default` and instead asks you to use `T::default`
```suggestion
        _ => (&[], node_comments.leading),
```

Reply:
```
error[E0308]: `match` arms have incompatible types
   --> crates/ruff_python_formatter/src/expression/mod.rs:241:14
    |
237 |       let (parentheses_comment, leading_inner) = match leading_inner.split_first() {
    |  ________________________________________________-
238 | |         Some((first, rest)) if first.line_position().is_end_of_line() => {
239 | |             (slice::from_ref(first), rest)
    | |             ------------------------------ this is found to be of type `(&[comments::SourceComment], &[comments::SourceComment])`
240 | |         }
241 | |         _ => (&[], node_comments.leading),
    | |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `(&[SourceComment], &[SourceComment])`, found `(&[_; 0], &[SourceComment])`
242 | |     };
    | |_____- `match` arms have incompatible types
    |
    = note: expected tuple `(&[comments::SourceComment], &[comments::SourceComment])`
               found tuple `(&[_; 0], &[comments::SourceComment])`
```

I'll revert to default